math/sum_of_digits.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2020 @author iamnambiar
+ * 
+ * @file
+ * A C++ Program to find the Sum of Digits of input integer.
+ */
+#include <iostream>
+#include <cassert>
+
+/**
+ * Function to find the sum of the digits of an integer.
+ * @param num The integer.
+ * @return Sum of the digits of the integer.
+ * 
+ * \detail
+ * First the algorithm check whether the num is negative or positive,
+ * if it is negative, then we neglect the negative sign.
+ * Next, the algorithm extract the last digit of num by dividing by 10 
+ * and extracting the remainder and this is added to the sum.
+ * The number is then divided by 10 to remove the last digit.
+ * This loop continues until num becomes 0. 
+ */
+int sum_of_digits(int num) {
```

Comment:
this means that the input number must be 32bit signed integer at most. Not an issue, and suffices to explain the algorithm.

Reply:
Thank you for the reply. Will fix the formatting issues you mentioned. Should I change the integer type to any other type?

Reply:
@kvedala I've updated the code.